src/Renci.SshNet/Security/Cryptography/Ciphers/AesGcmCipher.BclImpl.cs

```diff
@@ -0,0 +1,72 @@
+#if NET6_0_OR_GREATER
+using System;
+using System.Security.Cryptography;
+
+using Renci.SshNet.Common;
+using Renci.SshNet.Messages.Transport;
+
+namespace Renci.SshNet.Security.Cryptography.Ciphers
+{
+    internal partial class AesGcmCipher
+    {
+        private sealed class BclImpl : Impl
+        {
+            private readonly AesGcm _aesGcm;
+            private readonly int _tagSize;
+
+            public BclImpl(byte[] key, int tagSize)
+            {
+#if NET8_0_OR_GREATER
+                _aesGcm = new AesGcm(key, tagSize);
+#else
+                _aesGcm = new AesGcm(key);
+#endif
+                _tagSize = tagSize;
+            }
+
+            public override void Encrypt(byte[] nonce, byte[] input, int plainTextOffset, int plainTextLength, int associatedDataOffset, int associatedDataLength, byte[] output, int cipherTextOffset)
+            {
+                var cipherTextLength = plainTextLength;
+                var plainText = new ReadOnlySpan<byte>(input, plainTextOffset, plainTextLength);
+                var cipherText = new Span<byte>(output, cipherTextOffset, cipherTextLength);
+                var tag = new Span<byte>(output, cipherTextOffset + cipherTextLength, _tagSize);
+                var associatedData = new ReadOnlySpan<byte>(input, associatedDataOffset, associatedDataLength);
+
+                _aesGcm.Encrypt(nonce, plainText, cipherText, tag, associatedData);
+            }
+
+            public override void Decrypt(byte[] nonce, byte[] input, int cipherTextOffset, int cipherTextLength, int associatedDataOffset, int associatedDataLength, byte[] output, int plainTextOffset)
+            {
+                var plainTextLength = cipherTextLength;
+                var cipherText = new ReadOnlySpan<byte>(input, cipherTextOffset, cipherTextLength);
+                var tag = new ReadOnlySpan<byte>(input, cipherTextOffset + cipherTextLength, _tagSize);
+                var plainText = new Span<byte>(output, plainTextOffset, plainTextLength);
+                var associatedData = new ReadOnlySpan<byte>(input, associatedDataOffset, associatedDataLength);
+
+                try
+                {
+                    _aesGcm.Decrypt(nonce, cipherText, tag, output, associatedData);
+                }
+#if NET8_0_OR_GREATER
+                catch (AuthenticationTagMismatchException)
+#else
+                catch (CryptographicException ex) when (ex.Message == "The computed authentication tag did not match the input authentication tag.")
+#endif
+                {
+                    throw new SshConnectionException("MAC error", DisconnectReason.MacError);
+                }
```

Comment:
This seems pretty fragile, a) catching an undocumented exception and b) catching based on the (unlocalized) message text. I suggest just passing the caught exception as the inner exception

```suggestion
                catch (CryptographicException ex)
                {
                    throw new SshConnectionException("MAC error", DisconnectReason.MacError, ex);
                }
```

Reply:
`AuthenticationTagMismatchException` is a documented exception. See https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.authenticationtagmismatchexception?view=net-8.0

Reply:
It is not documented on the Encrypt/Decrypt methods (rightly or wrongly): https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.aesgcm.decrypt?view=net-8.0

Since AuthenticationTagMismatchException already derives from CryptographicException, I'm not sure there's any reason for the code to differ between .NET6/8

Reply:
Do we want to distinguish Mac Error and other error? See https://github.com/dotnet/runtime/issues/67082

Reply:
I don't think so - it's fatal for the connection either way

Reply:
Then it could fail with "Mac Error" where the underlying error is not "Mac Error".